### PR TITLE
Add Deribit Common exchange

### PR DESCRIPTION
**Pull request by artememelin**
### Pull request checklist

- [ ] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

File: src/Deribit/Utils.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+function Serde.deser(::Type{<:Data}, ::Type{<:Maybe{NanoDate}}, x::Int64)::NanoDate
+    return unixnanos2nanodate(x * 1e3)
+end
+
+function Serde.deser(::Type{<:Data}, ::Type{<:Maybe{NanoDate}}, x::AbstractString)::NanoDate
+    return unixnanos2nanodate(x * 1e3)
+end
+
+
```
Comment (yshadrin) [inline src/Deribit/Utils.jl:23]: extra enter

File: src/Deribit/Common/API/OrderBook.jl [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+Retrieves the order book, along with other market values for a given instrument.
+
+[`GET api/v2/public/get_order_book`](https://docs.deribit.com/#public-get_order_book)
+
+## Parameters:
+
+| Parameter | Type         | Required | Description |
+|:----------|:-------------|:---------|:------------|
+| instrument_name | String | true     |             |
+| depth           | Depth  | false    | `ONE` (1), `FIVE` (5), `TEN` (10), `TWENTY` (20), `FIFTY` (50), `ONE_HUNDRED` (100), `THOUSAND` (1000), `TEN_THOUSAND` (10000) |
```
Comment (yshadrin) [inline src/Deribit/Common/API/OrderBook.jl:100]: fix formatting

File: src/Deribit/Common/API/BookSummary.jl [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    future
+    option
+    future_combo
+    option_combo
+    spot
+end
+
+Base.@kwdef struct BookSummaryQuery <: DeribitPublicQuery
+    currency::Currency                       # BTC, ETH, USDC
+    kind::Maybe{InstrumentKind} = nothing    # option, spot, future_combo, option_combo
```
Comment (yshadrin) [inline src/Deribit/Common/API/BookSummary.jl:31]: add `future` in comment

File: src/Deribit/Common/API/Candle.jl [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+struct CandleData <: DeribitData
+    close::Maybe{Vector{Float64}}
+    cost::Maybe{Vector{Float64}}
+    high::Maybe{Vector{Float64}}
+    low::Maybe{Vector{Float64}}
+    open::Maybe{Vector{Float64}}
+    status::String
+    ticks::Maybe{Vector{NanoDate}}
+    volume::Maybe{Vector{Float64}}
```
Comment (yshadrin) [inline src/Deribit/Common/API/Candle.jl:46]: I think `nothing` can't come for these fields, at most an empty vector

File: src/Deribit/Common/API/Instrument.jl [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+[`GET api/v2/public/get_instruments`](https://docs.deribit.com/#public-get_instruments)
+
+## Parameters:
+
+| Parameter | Type           | Required | Description                                   |
+|:----------|:---------------|:---------|:----------------------------------------------|
+| currency  | Currency       | true     | `BTC` `ETH` `USDC` `USDT`                     |
+| expired   | Bool           | false    |                                               |
+| kind      | InstrumentKind | false    | `option` `spot` `future_combo` `option_combo` |
```
Comment (yshadrin) [inline src/Deribit/Common/API/Instrument.jl:78]: add `future`

File: src/Deribit/Common/API/OrderBook.jl [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    best_ask_amount::Float64
+    best_ask_price::Float64
+    best_bid_amount::Float64
+    best_bid_price::Float64
+    bid_iv::Maybe{Float64}
+    bids::Vector{Level}
+    change_id::Int64
+    current_funding::Maybe{Float64}
+    delivery_price::Maybe{Float64}
+    estimated_delivery_price::Float64
```
Comment (yshadrin) [inline src/Deribit/Common/API/OrderBook.jl:67]:
```suggestion
    estimated_delivery_price::Maybe{Float64}
```

File: src/Deribit/Common/API/OrderBook.jl [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    greeks::Maybe{Greeks}
+    index_price::Float64
+    instrument_name::String
+    interest_rate::Maybe{Float64}
+    last_price::Maybe{Float64}
+    mark_iv::Maybe{Float64}
+    mark_price::Float64
+    max_price::Float64
+    min_price::Float64
+    open_interest::Float64
```
Comment (yshadrin) [inline src/Deribit/Common/API/OrderBook.jl:78]:
```suggestion
    open_interest::Maybe{Float64}
```

File: src/Deribit/Common/API/Ticker.jl [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+Get ticker for an instrument.
+
+[`GET api/v2/public/ticker`](https://docs.deribit.com/#public-ticker)
+
+## Parameters:
+
+| Parameter | Type         | Required | Description |
+|:----------|:-------------|:---------|:------------|
+| instrument_name | String | true     |             |
```
Comment (yshadrin) [inline src/Deribit/Common/API/Ticker.jl:80]: fix formatting

File: src/Deribit/Deribit.jl [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+end
+
+"""
+    DeribitAPIError{T} <: AbstractAPIsError
+
+Exception thrown when an API method fails with code `T`.
+
+## Required fields
+- `jsonrpc::String`: The version of the JSON-RPC spec.
+- `reerror::DeribitAPIsErrorMsg`: Only present if there was an error invoking the method.
```
Comment (yshadrin) [inline src/Deribit/Deribit.jl:94]:
```suggestion
- `error::DeribitAPIsErrorMsg`: Only present if there was an error invoking the method.
```

**Review by yshadrin — APPROVED** [review]

File: src/Deribit/Utils.jl [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    return unixnanos2nanodate(x * 1e6)
+end
+
+function Serde.deser(::Type{<:DeribitData}, ::Type{<:Maybe{NanoDate}}, x::AbstractString)::NanoDate
+    return NanoDate(x)
+end
+
+function Serde.deser(::Type{<:DeribitData},::Type{<:Maybe{Vector{NanoDate}}}, x::Vector{Any})::Maybe{Vector{NanoDate}}
+    return unixnanos2nanodate.(x * 1e6)
+end
```
Comment (gryumov) [inline src/Deribit/Utils.jl:13]:
```suggestion
function Serde.deser(::Type{<:DeribitData}, ::Type{<:Maybe{Vector{NanoDate}}}, x::Vector{Any})::Maybe{Vector{NanoDate}}
    return unixnanos2nanodate.(x * 1e6)
end
```

**Review by gryumov — APPROVED** [review]

**Review by gryumov — APPROVED** [review]

**Pull request merged**